src/Glennsl__BsBenchmarkJs.re

```diff
@@ -0,0 +1,971 @@
+type benchmark;
+type suite;
+type suiteOptions;
+type deferred;
+type event;
+type options;
+type times;
+type stats;
+type platform;
+type support;
+type testFn = (. unit) => unit;
+type setupFn = (. unit) => unit;
+type teardownFn = (. unit) => unit;
+type eventHandler = (. event) => unit;
```

Comment:
I agree with you in general. I will go back and change that. At the time, I was concerned that the types might not be so uniform (much of Benchmark.js is unsoundly typed, and wildly polymorphic), and I was using those types everywhere. I was worried about keeping those types uniform across many files, which were later consolidated into a single file.

To be honest, I'm still not 100% sure that those functions are always strictly `unit => unit`, despite reading the source code many times over...

Reply:
So it looks like one of the reasons I created type aliases for these functions is that BuckleScript will throw a compiler error when using the `[@bs.get]` attribute when the return value is a function.

It complains about the arity of the external function:
```bash
  81 ┆ [@bs.get] external error: t => Js.Null_undefined.t(Js.Exn.t) = "error"
       ;
  82 ┆
  83 ┆ [@bs.get] external fn: t => unit => unit = "fn";
  84 ┆
  85 ┆ [@bs.get] external hz: t => float = "hz";

  Ill defined attribute [@bs.get] (only one argument)
```

The issue appears in...
```reason
// module Benchmark...
[@bs.get] external fn: t => (unit => unit) = "fn";
[@bs.get] external setup: t => unit => unit = "setup";
```
... and several others. All of these cases will fail compilation.

I have found two ways around this:
- returning an aliased type
- returning an explicitly uncurried function

I tried using `[@bs.send]`, but it will not actually return the function reference. Instead, it will return a curried function which wraps the execution of the target function.

This is not ideal.

For the test-case function, if a user wants to extract that test case and supply it to a new benchmark constructor, then the new test case will include the function wrapper, adding mysterious performance overhead.

For event-listener functions, it is important to keep an exact pointer to the listener function if you intend to remove that specific listener in the future. Functions that remove specific listeners will search the listener array using reference equality.

So maybe the alias is unnecessary, but I think it's probably better to at least use explicit uncurrying.

Reply:
Yeah, in this case it needs to be uncurried. Whenever a function is passed to or from JavaScript, it should be uncurried, but BuckleScript can often infer it. Not in this case though, since with currying a returned function is on the type-level indistinguishable from the function just taking another argument, as that's what currying is.

And if it's also unclear what the returned function does, then an alias helps document that. I don't think that's the case for this though, but it might be in other places.